### PR TITLE
Removed reference date and date type fields

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

**Comment by mickel1138** [comment]
Dataset entity also has Reference Year.

**Comment by mickel1138** [comment]
Also reference date in function clearDatasetSubmission in DatasetSubmissionController.php

**Comment by praneethpr** [comment]
> Also reference date in function clearDatasetSubmission in DatasetSubmissionController.php

Done

File: src/Entity/Dataset.php [hunk 1]
```diff
@@ -388,22 +388,6 @@ public function getAuthors()
         }
     }
 
-    /**
-     * Get reference date shortcut, to get Authors from DatasetSubmission.
-     *
-     * @return string|null
-     */
-    public function getReferenceDateYear()
```
Comment (mickel1138) [inline src/Entity/Dataset.php:396]: Removing this now may break other things until other tickets are merged.

Comment (praneethpr) [reply src/Entity/Dataset.php:396]: Yes, this pull request can be merged at the end

**Pull request merged**